data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/codec/PeerForwarderCodec.java

```diff
@@ -0,0 +1,7 @@
+package org.opensearch.dataprepper.peerforwarder.codec;
+
+public interface PeerForwarderCodec<T> {
+    byte[] serialize(T events) throws Exception;
```

Comment:
This need not be generic. It could take in a `Collection<Event>`. Then the implementations for each codec would be what knows the internal model.